pyquil/api/_qam.py

```diff
@@ -62,16 +63,22 @@ def load(self, executable):
     @_record_call
     def write_memory(self, *, region_name: str, offset: int = 0, value=None):
         """
-        Writes a value into a memory region on the QAM at a specified offset.
+        Writes a value or chronologically unwraps a list of values into a memory region on
+        the QAM at a specified offset.
 
         :param region_name: Name of the declared memory region on the QAM.
         :param offset: Integer offset into the memory region to write to.
-        :param value: Value to store at the indicated location.
+        :param value: Value(s) to store at the indicated location.
         """
         assert self.status in ['loaded', 'done']
 
-        aref = ParameterAref(name=region_name, index=offset)
-        self._variables_shim[aref] = value
+        if isinstance(value, List):
```

Comment:
```suggestion
        if isinstance(value, list):
```

I think you can just use the builtin type `list` here? if you make this change, you can also drop the `from typing import List`, above.

alternatively, consider switching this to a `collections.abc.Sequence` which would allow the caller to pass a `tuple` (or any other `Sequence`) instead. if you go this route, don't forget to update the docstring above to say "Sequence" rather than "list".